### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by asyriac**

### CODE REVIEW

File: src/utils/utils.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export const addItem = (arr, item) => {
+    // if (arr.find(element => element.product.id === item.id) === undefined) {
+    //     console.log(arr,item);
+        const res = arr.concat(item);
+        console.log(res);
+        return res;
+    // }
+    // console.log(arr,item);
+    // return arr;
```
Comment (Aravindan07) [inline src/utils/utils.js]: Remove the comments

File: src/components/ProductList.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { useProductContext } from "../contexts/product-context";
+import Product from "./Product";
+
+export const ProductList = () => {
+
+    const { loading, products, sortBy, excludeOutOfStock, showFastDeliveryOnly } = useProductContext();
+
+    console.log(products[0]);
```
Comment (Aravindan07) [inline src/components/ProductList.js]: Remove the log statements

File: src/components/WishlistItem.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        addItemToCart(item.product)
+    }
+
+    const handleRemoveFromWishlist = (item) => {
+        removeFromWishlist(item)
+    }
+
+    return (
+        <div className="card-h bg-white">
+            <img className="card-img-h" src={image} alt="" />
```
Comment (Aravindan07) [inline src/components/WishlistItem.js]: alt attribute is empty

**Review by Aravindan07 — CHANGES_REQUESTED** [review]
Hey, Arun Your application looks good. I have some changes to incorporate into your application.

1. Update the REAME.md file in Github.
2. I can add an item to the cart even though it is mentioned as out of stock. Fix this!
3.  Right now the wishlist icon appearing only on hover and also the icon size is too small. A user doesn't know if the particular item is on the wishlist or not if you plan to show the wishlist icon on hover. This is just a suggestion it's your wish to change it.
4.  Place the + button on the right and - button on the left you did the opposite. The reason to change this is that most of the users are right-handers so it would better if you place the - button on the right side.
5. It is showing the move to cart option for an item that is already added to the cart when that particular item is present in the wishlist. Fix this.
6.  You are allowing the same product to be added multiple times in the cart instead what you can do is increment the quantity of the product if it is added more than one time.
7. Change the title from React App to your application name.

File: src/App.js [hunk 4]
```diff
@@ -1,23 +1,16 @@
...
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div>
+      <Navbar />
+      <Route path="/" exact component={ProductListPage} />
+      <Route path="/cart" exact component={CartPage} />
+      <Route path="/wishlist" exact component={WishlistPage} />
```
Comment (sjsouvik) [inline src/App.js]: Add route for 404 Not found

**Review by sjsouvik — CHANGES_REQUESTED** [review]
Hi Arun,

The code looks clean, here're few suggestions from my end -

- remove default README
- add _redirects in public folder so that can prevent page not found error in netlify while you visit a route or refresh a route, let me know if you need more details on this
- add toasts to show some message on actions
- make it mobile responsive
- remove commented code and console log statements once the feature implementation is done

**Comment by khusharth** [comment]
- Loading can be placed in the center
- Filters UI can be better (optional)
- Can add search here
- Can use badge component from component library for fast delivery 
- https://flixcart.netlify.app/wishlist: There should be a wishlist heading on this page
- In the end of the website, Have some space or add a footer.
- On delete (optional): On delete have a confirmation modal.
- After any operation delete, add to card etc show an alert.
- Update the favicon on the project 
- Make Navbar, filter, cart page responsive.
- spacing can be improved in Price details on the Cart page

# BUG
- Move to cart button clears all wishlist

Optional features:
- Pagination

**Pull request merged**